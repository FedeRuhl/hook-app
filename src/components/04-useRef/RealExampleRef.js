import React, { useState } from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';
import './effects.css';

const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    const handleClick = () => setShow(!show);

    return (
        <div>
            <h1>Real example ref</h1>
            <hr />

            { (show) ? <MultipleCustomHooks /> : null }

            <button 
                className="btn btn-success"
                onClick={ handleClick }
            >
                Mostrar/ocultar
            </button>
        </div>
    )
};

export default RealExampleRef;