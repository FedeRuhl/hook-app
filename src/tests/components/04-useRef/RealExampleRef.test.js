import React from 'react';
import { shallow } from 'enzyme';
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';
// import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks';
// import useFetch from '../../../hooks/useFetch';
// import useCounter from '../../../hooks/useCounter';
// jest.mock('../../../hooks/useFetch');
// jest.mock('../../../hooks/useCounter');

describe('RealExampleRef tests', () => {
    test('should show the component succesfully', () => {
        const wrapper = shallow(<RealExampleRef />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });
    
    test('should show the component Multiple Custom Hooks', () => {
        const wrapper = shallow(<RealExampleRef />);
        let p = wrapper.find('p').text();
        expect(p).toBe('');
        
        const button = wrapper.find('button');
        button.simulate('click');
        
        p = wrapper.find('p').text();
        expect(p).toBe('<MultipleCustomHooks />');

        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
    
});