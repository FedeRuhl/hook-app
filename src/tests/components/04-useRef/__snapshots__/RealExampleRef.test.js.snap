// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef tests should show the component succesfully 1`] = `
<div>
  <h1>
    Real example ref
  </h1>
  <hr />
  <p />
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Mostrar/ocultar
  </button>
</div>
`;
